src/Model/ModelData.php

```diff
@@ -487,6 +491,52 @@ public function hasValue(string $field, array $arguments = [], bool $cache = tru
 
     // UTILITY METHODS -------------------------------------------------------------------------------------------------
 
+    /**
+     * Flags provides the user with additional data about the current page status.
+     *
+     * Mostly this is used for versioning, but can be used for other purposes (e.g. localisation).
+     * Each page can have more than one status flag.
+     *
+     * Returns an associative array of a unique key to a (localized) title for the flag.
+     * The unique key can be reused as a CSS class.
+     *
+     * Example (simple):
+     *   "deletedonlive" => "Deleted"
+     *
+     * Example (with optional title attribute):
+     *   "deletedonlive" => ['text' => "Deleted", 'title' => 'This page has been deleted']
+     */
+    public function getStatusFlags(bool $cached = true): array
+    {
+        if (!$this->_cache_statusFlags || !$cached) {
+            $flags = [];
+            $this->extend('updateStatusFlags', $flags);
+            $this->_cache_statusFlags = $flags;
+        }
+        return $this->_cache_statusFlags;
+    }
+
+    /**
+     * Get the HTML markup for rendering status flags for this model.
+     */
+    public function getStatusFlagMarkup(string $cssClasses = 'badge'): string
```

Comment:
Doing this instead of a template because it's run a _lot_ e.g. over every record in a gridfield, and rendering templates is relatively slow. This also retains the same code and functionality as previously, as closely as I can, which means less work for me right now since this is on the very edge of "in scope" for this card.